StockServiceFactory/src/main/java/edu/mbrooks/advancedjava/StockFactory/StockFactory.java

```diff
@@ -19,11 +21,17 @@
  */
 class BasicStockService implements StockService {
 
+    /**
+     * emum for which quarter a date is in
+     */
+    private enum Quarter {
+        Q1, Q2, Q3, Q4;
```

Comment:
not sure why you think this is needed.

Reply:
In the date range, I wanted to see what the quote was for the first recorded quote in every quarter. 

This not only makes perfect sense to me, I actually practice this, exactly for statistics I keep and report to our board of directors. Since this was familiar, right, wrong, or indifferent, the choice was mine, and therefore I chose exactly the intervals I'm familiar with for tracking statistics of issue types I continually track.